### PR TITLE
[StructuralMechanics] shells improving warning

### PR DESCRIPTION
**Pull request by philbucher**
now the warning should be printed only once

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_elements/shell_thick_element_3D4N.cpp [hunk 1]
```diff
@@ -504,24 +504,17 @@ void ShellThickElement3D4N::CalculateOnIntegrationPoints(const Variable<double>&
...
-				<< std::endl;
-		}
-	}
+    // The membrane formulation needs to iterate to find the correct
+    // mid-surface strain values.
+    // Check if we are doing a non-linear analysis type. If not, print warning
+
+    if (!rCurrentProcessInfo.Has(NL_ITERATION_NUMBER))
+    { // TODO should be KRATOS_WARNING_ONCE but this is currently not compiling
+        KRATOS_INFO_FIRST_N("ShellThickElement3D4N", 1) << "Gauss point results have "
```
Comment (armingeiser) [inline applications/StructuralMechanicsApplication/custom_elements/shell_thick_element_3D4N.cpp]: `KRATOS_WARNING_FIRST_N` does not exist am i right? Would fit better..

**Review by armingeiser — DISMISSED** [review]

**Review by MFusseder — DISMISSED** [review]

**Comment by philbucher** [comment]
@armingeiser the warning also exists, I just copied the wrong macro

**Review by armingeiser — DISMISSED** [review]

**Comment by philbucher** [comment]
@armingeiser I saw that there is another occurrence of the warning and added it

File: applications/StructuralMechanicsApplication/custom_elements/shell_thick_element_3D4N.cpp [hunk 2]
```diff
@@ -879,24 +879,17 @@ void ShellThickElement3D4N::CalculateOnIntegrationPoints(const Variable<Matrix>&
...
-				<< "\nThe membrane formulation used in the specified shell element"
-				<< "(ShellThickElement3D4N) requires iteration to accurately determine "
-				<< "recovered quantities (strain, stress, etc...).\n"
-				<< "Please switch to 'analysis_type = Non-Linear' in your json file for accurate recovered quantities."
-				<< std::endl;
-		}
-	}
+    if (!rCurrentProcessInfo.Has(NL_ITERATION_NUMBER))
+    { // TODO should be KRATOS_WARNING_ONCE but this is currently not compiling
+        KRATOS_WARNING_FIRST_N("ShellThickElement3D4N", 1) << "Warning: Gauss point results have "
```
Comment (armingeiser) [inline applications/StructuralMechanicsApplication/custom_elements/shell_thick_element_3D4N.cpp]: remove the Warning string

Comment (armingeiser) [reply applications/StructuralMechanicsApplication/custom_elements/shell_thick_element_3D4N.cpp]: and the newline

File: applications/StructuralMechanicsApplication/custom_elements/shell_thin_element_3D3N.cpp [hunk 3]
```diff
@@ -425,24 +425,17 @@ void ShellThinElement3D3N::CalculateOnIntegrationPoints(const Variable<Matrix>&
...
-				<< "\nThe membrane formulation used in the specified shell element"
-				<< "(ShellThinElement3D3N) requires iteration to accurately determine "
-				<< "recovered quantities (strain, stress, etc...).\n"
-				<< "Please switch to 'analysis_type = Non-Linear' in your json file for accurate recovered quantities."
-				<< std::endl;
-		}
-	}
+    if (!rCurrentProcessInfo.Has(NL_ITERATION_NUMBER))
+    { // TODO should be KRATOS_WARNING_ONCE but this is currently not compiling
+        KRATOS_WARNING_FIRST_N("ShellThinElement3D3N", 1) << "Warning: Gauss point results have "
```
Comment (armingeiser) [inline applications/StructuralMechanicsApplication/custom_elements/shell_thin_element_3D3N.cpp:434]: same

**Review by armingeiser — CHANGES_REQUESTED** [review]
I just saw that the message sometimes includes "Warning" and a "/n" - sometimes not. I would make them consistent somehow, however you like.

EDIT: my mistake, everything was fine...

**Review by armingeiser — APPROVED** [review]

**Pull request merged**